### ISSUE TITLE
data.strip() return 'Failure: 200 OK'

### ISSUE THREAD

**Issue by roiff** [pkkid/python-plexapi#326]
homeassistant control players return error:
```
Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/Users/chenyun/Projects/home-assistant/lib/python3.6/site-packages/homeassistant/helpers/service.py", line 277, in _handle_service_platform_call
    await getattr(entity, func)(**data)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/chenyun/Projects/home-assistant/lib/python3.6/site-packages/homeassistant/components/media_player/plex.py", line 716, in media_play
    self.device.play(self._active_media_plexapi_type)
  File "/Users/chenyun/Projects/home-assistant/lib/python3.6/site-packages/plexapi/client.py", line 308, in play
    self.sendCommand('playback/play', type=mtype)
  File "/Users/chenyun/Projects/home-assistant/lib/python3.6/site-packages/plexapi/client.py", line 196, in sendCommand
    return self.query(key, headers=headers)
  File "/Users/chenyun/Projects/home-assistant/lib/python3.6/site-packages/plexapi/client.py", line 168, in query
    return ElementTree.fromstring(data) if data.strip() else None
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/xml/etree/ElementTree.py", line 1314, in XML
    parser.feed(text)
  File "<string>", line None
xml.etree.ElementTree.ParseError: syntax error: line 1, column 0
```
I tried to view the data of `data.strip()` and returned: 
```
b'Failure: 200 OK'
```

**Comment by Hellowlol** [comment]
Is this still a issue? It seems that response you got was not a proper xml.

**Comment by jjlawren** [comment]
I'm seeing this error consistently when using `client.playMedia()` to play music on a Plexamp device. The media plays successfully, but this exception is not caught.

Related issue: https://github.com/home-assistant/home-assistant/issues/19623

**Comment by Hellowlol** [comment]
I’m not sure i want to catch the syntax error. Imo the clients should send a proper response. Our client testing is really lacking as the api is undocumented and catching the error might not be correct for other clients then plexamp

**Comment by jjlawren** [comment]
I'd consider Plexamp an "official" client as Plex actively promotes it. Too bad it's misbehaving. I'll catch the error myself for now.

**Comment by jjlawren** [comment]
Okay, gave it a shot but catching outside of the library isn't scalable. Every command sent to the client would need to be wrapped. I know it's not ideal to catch this error unconditionally, but perhaps there's another check that can be added? It seems like it's an empty response.

**Comment by pkkid** [comment]
Empty responses should be handled properly by plexapi imo. They are pretty common.

If it's malformed JSON, then we shouldn't handle it. We made a conscious decision early on not to write specific code to handle all the various clients sending us different information. It would get unwieldy too quick.

**Comment by jjlawren** [comment]
Ah, that's a good point. I'll dig some more over the weekend.

If the contents is not empty but is still invalid XML, does it make sense to catch and log an error (and perhaps raise a more useful exception)?

**Comment by jjlawren** [comment]
> Perhaps we just treat this specific string as an empty response `Failure: 200 OK`?

I like that.

**Comment by Hellowlol** [comment]
Let’s report this error upstream, if they dont fix it. We can catch and log the error just for for plexamp.

**Comment by pjv** [comment]
Plex never fixes errors. They are far too busy developing half-assed features that nobody wants.

**Comment by jjlawren** [comment]
Note that this same bad response is caught in this `sendCommand` call: https://github.com/pkkid/python-plexapi/blob/9c50225df8ee4bfe95695e8d9655c0277a8ee489/plexapi/client.py#L463-L469

This exception is logged and passed along to containing applications. Perhaps this log can be moved to debug level as it's not useful for the user?

**Comment by jjlawren** [comment]
Dug around a bit and Plexamp is responding with... `b'OK'`. Which I guess is marginally better than `b'Failure: 200 OK'`, but still annoying.

I agree that this should be fixed upstream, but checking for a specific improperly formatted success message seems like a reasonable fix in the meantime.

Created https://github.com/pkkid/python-plexapi/pull/387 if you agree.

**Comment by Hellowlol** [comment]
The other bug you refererred. I want to keep it as it is. See the comment for why its done.

About your fix. Looks god, but I’d prefer that we handle this at the highest level we can (inside sendCommend and only if its plexamp

**Comment by jjlawren** [comment]
> The other bug you refererred. I want to keep it as it is. See the comment for why its done.
Definitely. It was just logging an exception I could not catch/handle from outside the library.

I'll update the PR to make the fix more targeted.

**Comment by Hellowlol** [comment]
Why do you need to catch/handle the exception? Anyway of you really need it just change to loglevel to debug and include the stacktracd.

**Comment by jjlawren** [comment]
I'm saying that if I call the function as so:
```
try:
    client.playMedia(media)
except xml.etree.ElementTree.ParseError:
    pass
```
then the code in [this comment](https://github.com/pkkid/python-plexapi/issues/326#issuecomment-538666087) will still log the exception in full when subscribing to the timeline. It can't be fully handled/caught from outside the library.

**Comment by jjlawren** [comment]
I don't see a clean way to handle this in `sendCommand()` besides a `try:` clause since the call to `method()` and the parsing of its response is all handled within `query()`. I think it has to be done in `query()` since the current logic attempts to XML parse every non-empty response from a client.

Open to other ideas, of course.

**Comment by Hellowlol** [comment]
Thats fine, its what i wanted.

**Comment by jjlawren** [comment]
Updated PR (#387) based on suggestions.

**Comment by jjlawren** [comment]
This should be fixed by #387 and #394. Other clients that behave like this can be patched the same way in future PRs.
